app/views/users/registrations/new.html.erb

```diff
@@ -11,9 +13,8 @@
 
     <div class="govuk-grid-column-two-thirds govuk-!-padding-left-0">
       <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
-        <%= devise_error_messages! %>
 
-        <div class="govuk-form-group">
+        <div class="govuk-form-group <%= "govuk-form-group--error" if resource&.errors&.any? %>">
```

Comment:
Wonder if there is any way to remove the duplication of this, can't think of anything off the top of my head.

Reply:
Short of creating a longwinded form helper for this one style, I'm also unsure.

Reply:
Would love to create something like this, but I'm unsure how to get it working:
```
        <%= gov_form_group_tag do |f, resource| %>
          <%= f.label :email, "Provide your email address", class: "govuk-label" %><br />
          <%= f.email_field :email, autofocus: true, autocomplete: "email", class: "govuk-input",
            value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email), required: true %>
        <% end %>
```
```
  def gov_form_group_tag(&block)
    classes = ["govuk-form-group"]
    classes << "govuk-form-group--error" if resource&.errors&.any?
    content_tag :div, class: classes  do
      yield
    end
  end
```

Reply:
I had a go but abandoned this approach as it led to me having to use `form_tags` and deviate quite far from a simple rails form. All for a dynamic class.